src/test/java/org/mskcc/oncokb/curation/service/CancerTypeServiceIT.java

```diff
@@ -0,0 +1,113 @@
+import org.junit.jupiter.api.Test;
+import org.mskcc.oncokb.curation.IntegrationTest;
+import org.mskcc.oncokb.curation.domain.CancerType;
+import org.mskcc.oncokb.curation.domain.RelevantCancerTypeBody;
+import org.mskcc.oncokb.curation.domain.RelevantCancerTypeQuery;
+import org.mskcc.oncokb.curation.repository.CancerTypeRepository;
+import org.mskcc.oncokb.curation.util.enumeration.RelevantCancerTypeDirection;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@IntegrationTest
+public class CancerTypeServiceIT {
+
+    @Autowired
+    private CancerTypeService cancerTypeService;
+
+    @Autowired
+    private CancerTypeRepository cancerTypeRepository;
+
+    @Test
+    public void testGetRelevantCancerTypesWithoutExcluded() throws Exception {
+        RelevantCancerTypeQuery included = new RelevantCancerTypeQuery();
+        included.setCode("MCD");
+        included.setMainType("Mastocytosis");
+
+        RelevantCancerTypeBody relevantCancerTypeBody = new RelevantCancerTypeBody();
+        relevantCancerTypeBody.setRelevantCancerTypeQueries(Arrays.asList(included));
+
+        List<CancerType> cancerTypes = cancerTypeService.getRelevantCancerTypes("Level_1", relevantCancerTypeBody);
+        String expectedResult =
+            "Mastocytosis, Cutaneous Mastocytosis, Mast Cell Sarcoma, Systemic Mastocytosis, Aggressive Systemic Mastocytosis, Smoldering Systemic Mastocytosis, Mast Cell Leukemia, Systemic Mastocytosis with an Associated Hematological Neoplasm, Indolent Systemic Mastocytosis";
+        assertEquals(expectedResult, cancerTypesToString(cancerTypes));
+    }
+
+    @Test
+    public void testGetRelevantCancerTypesWithExcluded() {
+        RelevantCancerTypeQuery included = new RelevantCancerTypeQuery();
+        included.setCode("MCD");
+        included.setMainType("Mastocytosis");
+        RelevantCancerTypeQuery excluded = new RelevantCancerTypeQuery();
+        excluded.setCode("SM");
+        excluded.setMainType("Mastocytosis");
+
+        RelevantCancerTypeBody relevantCancerTypeBody = new RelevantCancerTypeBody();
+        relevantCancerTypeBody.setRelevantCancerTypeQueries(Arrays.asList(included));
+        relevantCancerTypeBody.setExcludedRelevantCancerTypeQueries(Arrays.asList(excluded));
+
+        List<CancerType> cancerTypes = cancerTypeService.getRelevantCancerTypes("Level_1", relevantCancerTypeBody);
+        String expectedResult = "Mastocytosis, Cutaneous Mastocytosis, Mast Cell Sarcoma"; // excluding also excludes main type if the same. Do we want this?
+        assertEquals(expectedResult, cancerTypesToString(cancerTypes));
+    }
+
+    @Test
+    public void testFindCancerTypesWithDirection() throws Exception {
+        List<CancerType> cancerTypes = cancerTypeService.findRelevantCancerTypes("CML", false, RelevantCancerTypeDirection.DOWNWARD);
+        String expectedResult = "M:All Liquid Tumors, M:All Tumors, Chronic Myelogenous Leukemia, Chronic Myeloid Leukemia, BCR-ABL1+";
```

Comment:
I do expected to see matched main type to be sorted at the end after the matched cancer type. Is it not the case?

Reply:
Since we decided to sort by level, main types will end up at the front. However, we can start with the matched cancer type and then sort by level.